test/e2e/tests/settings/settings-security-reveal-srp.spec.js

```diff
@@ -26,11 +26,12 @@ describe('Reveal SRP through settings', function () {
         await completeSRPRevealQuiz(driver);
         await driver.fill('#password-box', wrongTestPassword);
         await driver.press('#password-box', driver.Key.ENTER);
-        const passwordErrorIsDisplayed = await driver.isElementPresent({
+        await driver.isElementPresent({
           css: '.mm-help-text',
-          text: 'Incorrect password',
         });
-        assert.equal(passwordErrorIsDisplayed, true);
+        const displayedError = await driver.findVisibleElement('.mm-help-text');
+        const displayedErrorMessage = await displayedError.getText();
```

Comment:
This could make the test more flaky, as we're introducing two new async steps in between finding the element and asserting. The element could be found, then removed from the DOM or had its contents changed.

Reply:
While true, if this is the case, wouldn't that be indicative of an actual error we do want to fail (ideally consistently, since that should never happen right?)

Reply:
It's true that the type of page behavior that would trigger flakiness in this case is probably undesirable.

But we do have UI that is meant to refresh (e.g. gas estimates, balances, the activity list, etc.). And we have a lot of janky UI that rerenders when it shouldn't, causing flakiness in cases like this, where it would be helpful for the test to assert _something_ about the UI behavior even in the face of this problem.

Hmm. Tricky problem. Maybe you're right that having tests that are less-tolerant of jankyness would encourage us to fix the jankyness. But given the current state of our e2e test suite, my intuition is that it would do more harm than good in the short/medium term because we don't have capacity to do a good job of fixing janky UI everywhere right now, and we desperately need our tests to be less flaky. But I am definitely open to being convinced otherwise here.